XenModel/DockerContainers.cs

```diff
@@ -106,20 +106,20 @@ private static void CollectionChanged(Object sender, CollectionChangeEventArgs e
         {
             InvokeHelper.AssertOnEventThread();
 
-            IXenObject ixmo = e.Element as IXenObject;
-            if (ixmo == null)
+            VM vm = e.Element as VM;
+            if (vm == null)

```

Comment:
Is it the case that only VMs can ever reach here? If so, it would be better if this were an Assert?